applications/common/backend/src/test/java/pl/ds/kyanite/common/components/models/table/TableColspansCalculationTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2024 Dynamic Solutions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pl.ds.kyanite.common.components.models.table;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.*;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(SlingContextExtension.class)
+public class TableColspansCalculationTest {
+
+  private final SlingContext context = new SlingContext(ResourceResolverType.RESOURCERESOLVER_MOCK);
+
+  @BeforeEach
+  public void init() {
+    context.addModelsForClasses(TableCellComponent.class, TableHeadCellComponent.class, TableComponent.class);
+  }
+
+  private static Stream<Arguments> inputColumnsAndExpectedComputedColspans() {
+    return Stream.of(
+            Arguments.of(
+                    columns(4, 4, 4, null),
+                    colspans(4, 4, 4, 1)),
+            Arguments.of(
```

Comment:
This list of arguments, is actually the documentation of the change.
The pairs are:
 - list of `columns` property value for all cells in given table row
 - and the result computed colspans of the same cells